clang/lib/Lex/Preprocessor.cpp

```diff
@@ -887,6 +906,9 @@ void Preprocessor::Lex(Token &Result) {
     switch (CurLexerKind) {
     case CLK_Lexer:
       ReturnedToken = CurLexer->Lex(Result);
+      if (ReturnedToken && CurLexer &&
+          CurLexer->getFileID() == SourceMgr.getMainFileID())
+        saveCheckPoint(CurLexer->BufferPtr);
```

Comment:
Should we test for `IsAtStartOfLine` here?


```suggestion
      if (ReturnedToken &&
          CurLexer->getFileID() == SourceMgr.getMainFileID())
        saveCheckPoint(CurLexer->BufferPtr);
```

Reply:
The check for `CurLexer` is needed in case the call to `Lex` above unsets it.

Reply:
can you benchmark after adding `IsAtStartOfLine` ?

Reply:
That would be http://llvm-compile-time-tracker.com/compare.php?from=6ade5183232dc1398205d7c9dbe21243b2560837&to=5ad516d39b6706af6fd55eded264aba4ba432ad2&stat=instructions:u